src/postgraphile/__tests__/postgraphileIntegrationShutdown-test.js

```diff
@@ -0,0 +1,29 @@
+jest.unmock('postgraphile-core');
+
+import pgPool, { poolConfig } from '../../__tests__/utils/pgPool';
+import { postgraphile } from '..';
+
+const COMMON_OPTIONS = {
+  exitOnFail: false,
+};
+
+test('When the handler is created using a Pool object, it can be released without triggering an error', async () => {
+  let handler = postgraphile(pgPool, COMMON_OPTIONS);
+  await handler.release();
```

Comment:
```suggestion
  let handler = postgraphile(pgPool, COMMON_OPTIONS);
  await handler.release();
  // This would throw an error if the pool was released, it not throwing is what passes the test.
  await pgPool.query("select 'Pool is still valid';");
```